libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -701,14 +707,18 @@ internal class MapboxTripSession(
     }
 
     /**
-     * Invalidate latest banner instruction. To get the latest banner instruction wrapper call
-     * [BannerInstructionEvent.latestInstructionWrapper]
+     * Invalidate latest banner and voice instruction. To get the latest banner instruction wrapper call
+     * [BannerInstructionEvent.latestInstructionWrapper], to get the latest voice instruction
+     * call [lastVoiceInstruction]
      */
     private fun invalidateLatestInstructions(
-        latestInstructionWrapper: BannerInstructionEvent.LatestInstructionWrapper?
+        latestInstructionWrapper: BannerInstructionEvent.LatestInstructionWrapper?,
+        voiceInstruction: VoiceInstructions?,
     ) {
         bannerInstructionEvent.invalidateLatestBannerInstructions(latestInstructionWrapper)
-        lastVoiceInstruction = null
+        if (lastVoiceInstruction == voiceInstruction) {
```

Comment:
I think we now have a case when the index won't be set to null but the voice instruction will.
The index will be null if `latestInstructionWrapper` is equal to saved one. But `latestInstructionWrapper` is about banner instruction and index. So can there be the following case:
1) banner instruction is updated;
2) index is not updated;
3) voice instruction is not updated;
4) we invoke `invalidateLatestInstructions`;
5) result: banner instruction is not null, voice instruction is null?

I'm not sure, maybe this just can't happen. just double-checking.

Reply:
that actually a good point. It seems not critical for the voice instruction as it's pronounced once. 
Let's cut a ticket to address the issue separately. It looks like refactoring task, along with https://github.com/mapbox/mapbox-navigation-android/pull/6689/files#r1041952357